### ISSUE TITLE
nextcloud docker nginx traefik error after upgrade nextcloud 21 - /.well-known/webfinger - /.well-known/nodeinfo

### ISSUE THREAD

**Issue by RedJohn14** [nextcloud/docker#1438]
Hello,

After I have upgraded my nextcloud (image: nextcloud:fpm-alpine) I have this error messages:

**Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/webfinger" aufzulösen.
Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/nodeinfo" aufzulösen.**

I have checked my **nginx.conf** and I used this one here from github ( https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/postgres/fpm/web/nginx.conf )

I used traefik 2.0 to proxy connections to nginx backend, traefik 2.0 config / labels looks like this:

```
      - "traefik.http.routers.nextcloud-app-secure.middlewares=nextcloud-dav,secHeaders@file"
      - "traefik.http.middlewares.nextcloud-dav.replacepathregex.regex=^/.well-known/ca(l|rd)dav"
      - "traefik.http.middlewares.nextcloud-dav.replacepathregex.replacement=/remote.php/dav/"
```

Config from here:

https://goneuland.de/nextcloud-server-mit-docker-compose-und-traefik-installieren/

How can I fix this problem I have no idea ... :-(

**Comment by ThxAndBye** [comment]
Your Nginx config needs to be updated according to the docs:  
https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html

**Comment by RedJohn14** [comment]
> Your Nginx config needs to be updated according to the docs:
> https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html

I have used this new config, but have still the error.

Any ideas why?

**Comment by RedJohn14** [comment]
> Your Nginx config needs to be updated according to the docs:
> https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html

@ThxAndBye I have double check my nginx config here is my config, any mistake?

This is my config:

https://pastebin.com/SEJBaaDW


I have also used this config here: https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html but have always the message in nextcloud. I used nginx.

**Comment by ThxAndBye** [comment]
I'm using the config from the docs (just edited the ssh certificates and php upstream) and it passes all checks for me.  
As redirects need to be handled differently it's necessary to use the new nginx configuration for NC21.
![image](https://user-images.githubusercontent.com/16294365/110854170-d5587380-82b4-11eb-9525-d55c6ebe9de8.png)

**Comment by RedJohn14** [comment]
> I'm using the config from the docs (just edited the ssl certificates and php upstream) and it passes all checks for me.
> As redirects need to be handled differently it's necessary to use the new nginx configuration for NC21.
> ![image](https://user-images.githubusercontent.com/16294365/110854170-d5587380-82b4-11eb-9525-d55c6ebe9de8.png)

@ThxAndBye Okay but I used nextcloud inside docker container with docker image **nextcloud:fpm-alpine** and some **nginx** container. I also used some **traefik** **proxy**.

I have also this error here:
https://github.com/nextcloud/docker/issues/1414

Did you also use docker image with nginx (with or without traefik) or direct install nextcloud on some server?

**Comment by ThxAndBye** [comment]
I'm using the nextcloud:fpm and nginx:stable containers.

**Comment by RedJohn14** [comment]
> I'm using the nextcloud:fpm and nginx:stable containers.

@ThxAndBye okay but you not use the alpine image right? Can you test with the **nextcloud:fpm-alpine** and check if you get this error here? https://github.com/nextcloud/docker/issues/1414

Did you also use traefik?

**Comment by ThxAndBye** [comment]
I'd get the same error as #1414 if I wouldn't fix it manually after container start.  
No I use nginx:stable and certbot/dns-cloudflare:latest for encryption. I want to use as little additional software as possible. Traefik would only add an additional layer.

**Comment by RedJohn14** [comment]
> I'd get the same error as #1414 if I wouldn't fix it manually after container start. (as a stopgap until it's fixed)
> No I use nginx:stable and certbot/dns-cloudflare:latest for encryption. I want to use as little additional software as possible. Traefik would only add an additional layer.

@ThxAndBye Okay thanks for fast reply. Now I have edit my **nginx.conf** with new settings from nextcloud docs. BUT I have also this error message but why?

Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/webfinger" aufzulösen. Weitere Informationen findest Du in der Dokumentation.
Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/nodeinfo" aufzulösen. Weitere Informationen findest Du in der Dokumentation.

Any idea where the mistake is? I have no idea at the moment.

For information, I have nginx configure without ssl certificate only on port 80 because traefik is my ssl proxy with letsencrypt. I have this error after upgrade to nextcloud 21.

**Comment by ThxAndBye** [comment]
Have you reloaded Nginx? E.g. by restarting the container or executing `nginx -s reload` in the Nginx container?  
If you are sure that the new config is applied then it might be related to Traefik and as I don't have experience with that so I can't help you in that regard.

**Comment by RedJohn14** [comment]
> Haben Sie die Nginx-Konfiguration neu geladen? ZB durch Neustart des Containers oder Ausführen `nginx -s reload`im Nginx-Container?
> Wenn Sie sicher sind, dass die neue Konfiguration angewendet wird, hängt sie möglicherweise mit Traefik zusammen. Da ich keine Erfahrung damit habe, kann ich Ihnen in dieser Hinsicht nicht helfen.

@ThxAndBye Yes new config is applied but error messages are still in nextcloud. Okay no problem, maybe someone have information about traefik updated config?!

**Comment by RedJohn14** [comment]
> Have you reloaded the Nginx config? E.g. by restarting the container or executing `nginx -s reload` in the Nginx container?
> If you are sure that the new config is applied then it might be related to Traefik and as I don't have experience with that so I can't help you in that regard.

@ThxAndBye Okay I have new nginx config but still with the same error, but I have found new information with browser debug:

https://github.com/nextcloud/docker/issues/1443

Maybe some idea?

**Comment by Tockra** [comment]
I'm using an docker installation with nginx and maria db.
I took these settings:
https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb/fpm/web/nginx.conf
I have the same issue as RedJohn14. Did somebody find an solution?
![grafik](https://user-images.githubusercontent.com/1353859/132130158-65c8c423-cfbd-411a-b70a-5fc3dac009ea.png)

**Comment by gibald** [comment]
I think this issue is duplicate, the solution of the other one is here : https://github.com/nextcloud/docker/issues/1443#issuecomment-799541831
